x-pack/plugins/ml/public/timeseriesexplorer/timeseries_chart.js

```diff
@@ -0,0 +1,1239 @@
+  getMultiBucketImpactLabel,
+} from '../../common/util/anomaly_utils';
+import { formatValue } from '../formatters/format_value';
+import {
+  LINE_CHART_ANOMALY_RADIUS,
+  MULTI_BUCKET_SYMBOL_SIZE,
+  SCHEDULED_EVENT_SYMBOL_HEIGHT,
+  drawLineChartDots,
+  filterAxisLabels,
+  numTicksForDateFormat,
+  showMultiBucketAnomalyMarker,
+  showMultiBucketAnomalyTooltip,
+} from '../util/chart_utils';
+import { TimeBuckets } from 'ui/time_buckets';
+import { mlAnomaliesTableService } from '../components/anomalies_table/anomalies_table_service';
+import ContextChartMask from './context_chart_mask';
+import { findChartPointForAnomalyTime } from './timeseriesexplorer_utils';
+import { mlEscape } from '../util/string_utils';
+import { mlFieldFormatService } from '../services/field_format_service';
+import { mlChartTooltipService } from '../components/chart_tooltip/chart_tooltip_service';
+
+const focusZoomPanelHeight = 25;
+const focusChartHeight = 310;
+const focusHeight = focusZoomPanelHeight + focusChartHeight;
+const contextChartHeight = 60;
+const contextChartLineTopMargin = 3;
+const chartSpacing = 25;
+const swimlaneHeight = 30;
+const margin = { top: 20, right: 10, bottom: 15, left: 40 };
+
+const ZOOM_INTERVAL_OPTIONS = [
+  { duration: moment.duration(1, 'h'), label: '1h' },
+  { duration: moment.duration(12, 'h'), label: '12h' },
+  { duration: moment.duration(1, 'd'), label: '1d' },
+  { duration: moment.duration(1, 'w'), label: '1w' },
+  { duration: moment.duration(2, 'w'), label: '2w' },
+  { duration: moment.duration(1, 'M'), label: '1M' }];
+
+// Set up the color scale to use for indicating score.
+const anomalyColorScale = d3.scale.threshold()
+  .domain([3, 25, 50, 75, 100])
+  .range(['#d2e9f7', '#8bc8fb', '#ffdd00', '#ff7e00', '#fe5050']);
+
+// Create a gray-toned version of the color scale to use under the context chart mask.
+const anomalyGrayScale = d3.scale.threshold()
+  .domain([3, 25, 50, 75, 100])
+  .range(['#dce7ed', '#b0c5d6', '#b1a34e', '#b17f4e', '#c88686']);
+
+function getSvgHeight() {
+  return focusHeight + contextChartHeight + swimlaneHeight + chartSpacing + margin.top + margin.bottom;
+}
+
+export class TimeseriesChart extends React.Component {
+  static propTypes = {
+    autoZoomDuration: PropTypes.number,
+    contextAggregationInterval: PropTypes.object,
+    contextChartData: PropTypes.array,
+    contextForecastData: PropTypes.array,
+    contextChartSelected: PropTypes.func,
+    detectorIndex: PropTypes.string,
+    focusAggregationInterval: PropTypes.object,
+    focusChartData: PropTypes.array,
+    focusForecastData: PropTypes.array,
+    modelPlotEnabled: PropTypes.bool,
+    renderFocusChartOnly: PropTypes.bool,
+    selectedJob: PropTypes.object,
+    showForecast: PropTypes.bool,
+    showModelBounds: PropTypes.bool,
+    svgWidth: PropTypes.number,
+    swimlaneData: PropTypes.array,
+    timefilter: PropTypes.object,
+    zoomFrom: PropTypes.object,
+    zoomTo: PropTypes.object
+  }
+
```

Comment:
Might be useful to add `isRequired` to these to make it easier/faster to catch missing props or any issues with props being passed in incorrectly.

Reply:
Thx, added a note to #25791 to do that in a follow-up.